application.py

```diff
@@ -374,6 +375,63 @@ def make_payload(data, errormessage="", errorcode=200):
     }), errorcode)
 
 
+def expression_matrix_csv(output_cellset, filename="cxg_expression_matrix.csv"):
+    """
+    Creates csv download for a cellset
+    :param output_cellset: name of cellset with data
+    :return: Flask response containing csv of expression matrix
+    """
+    global e
+    expression = e.getDenseExpressionMatrix("AllGenes", output_cellset)
+    expression = expression.transpose()
+    genes = all_genes()
+    cellids = e.getCellSet(output_cellset)
+
+    def generate_expression_matrix_csv(expression, genes, cellids):
+        first_row = [""] + [get_cell_name(cid) for cid in cellids]
+        yield ",".join(first_row) + "\n"
+        for idx, row in enumerate(expression):
+            line = [genes[idx]] + row.tolist()
+            yield ",".join(str(e) for e in line) + "\n"
+
+    headers = Headers()
+    headers.add("Content-Disposition", "attachment; filename={}".format(filename))
+    return Response(generate_expression_matrix_csv(expression, genes, cellids), headers=headers, mimetype='text/csv')
+
+
+def metadata_csv(cell_ids, filename="cxg_metadata.csv"):
+    """
+    Creates csv download for a cellset
+    :param output_cellset: name of cellset with data
+    :return: Flask response containing csv of expression matrix
+    """
+    global e
+    metadata = parse_metadata(cell_ids)
+
+    def metadata2table(metadata):
+        """
+        Generator to turn results of parse_metadata to a table of metadata values with rows being cells
+        and columns being metadata fields
+        :param metadata: metadata from cell set produced by parse_metadata
+        :return: 2d list of metadata values with header
+        """
+        if not len(metadata):
+            return
+        header = list(metadata[0].keys())
+        header.sort()
+        # put cell name in front
+        header.remove("CellName")
+        header = ["CellName"] + header
+        yield ",".join(header) + "\n"
```

Comment:
Need to ensure that the values in a CSV are escaped properly (consider what will happen if a value contains a comma or some other magic character).

You may consider using a CSV writer class (eg, https://docs.python.org/3.6/library/csv.html) which will do this automatically.  Alternatively will need to implement proper escaping (see eg, https://tools.ietf.org/html/rfc4180)